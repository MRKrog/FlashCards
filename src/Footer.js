import React from "react";

import './styles/Footer.scss';

const Footer = props => {
  if(props.currentPlayer) {
    return (
      <footer className="App-footer">
        <div className="BtnAnswer">
          <button onClick={props.resetQuiz}>Reset Game</button>
        </div>
      </footer>
    );
  }
  return (
    <footer className="App-footer">
      <p>&copy; 2019</p>
    </footer>
  );

};

export default Footer;
