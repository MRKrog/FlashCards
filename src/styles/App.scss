// font-family: 'Rajdhani', sans-serif;
// font-family: 'Lato', sans-serif;

// Segoe UI", "Roboto", "Oxygen",
  // "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",

$font-title: 'Abel', sans-serif;
$font-header: 'Rajdhani', sans-serif;
$font-copy: 'Lato', sans-serif;

$white: #fff;
$black: #231f20;
$dark-blue: #11212a;

.App {
  background:url(code_bg.jpg);
  width: 100%;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  z-index: 9;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, .4);
    background: radial-gradient(ellipse at center, rgba(76,76,76,1) 0%, rgba(17,17,17,0.77) 51%, rgba(19,19,19,0.55) 100%);
  }

  .App-header {
    background-color: #231f20;
    display: block;
    width: 100%;
    z-index: 10;
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, .8);

    h1 {
      font-family: $font-title;
      color: #4c4c4c;
      text-transform: uppercase;
      font-size: 2em;
      letter-spacing: .1em;
      text-align: center;
      margin: 0;
      padding: .2em;
      line-height: 1;
      font-weight: 100;
    }

  }

  .main-content {
    width: 100%;
    z-index: 10;

    .welcome-btn {
      width: 100%;
      display: block;
      margin: auto;
      max-width: 200px;
      font-family: "Lato", sans-serif;
      transition: all .5s ease;
      font-size: 1em;
      border: none;
      border-radius: 5px;
      padding: .4em;
      cursor: pointer;
      background: #1f93ec;
      color: #fff;
      outline: none;
    }
  }

  .App-footer {
    background-color: #231f20;
    display: block;
    width: 100%;
    z-index: 10;
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, .8);

    p {
      margin: 0;
      text-align: center;
      color: #4c4c4c;
      font-family: $font-title;
      padding: .5em
    }
  }

}
