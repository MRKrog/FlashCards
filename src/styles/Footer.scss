$font-title: 'Abel', sans-serif;
$lightWhite: #d6d6d6;

.App-footer {
  display: block;
  z-index: 10;
  padding: .6em;

  p {
    margin: 0;
    text-align: center;
    color: $lightWhite;
    font-family: $font-title;
    padding: .5em
  }

  .BtnAnswer {
    padding: 0;
    text-align: right;

    button {
      width: 100px;
      font-size: .8em;
      padding: .5em;
      box-shadow: none;
      text-transform: uppercase;
      border: none;
      background: rgba(77, 27, 169, 0.5);
      color: rgba(255, 255, 255, .6);
    }
  }
}
