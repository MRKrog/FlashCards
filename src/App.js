import React, { Component } from 'react';

import Header from './Header.js';
import WelcomeScreen from './WelcomeScreen.js';
import CardContainer from './CardContainer.js';
import Footer from './Footer.js';

import './styles/App.scss';

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentPlayer: false,
      originCards: [],
      allCards: [],
      correctCards: [],
      currCardIndex: 0,
      localStorageExist: false,
      cardsLeft: 0
    }
  }

  componentDidMount() {
    fetch("http://memoize-datasets.herokuapp.com/api/v1/mkCards")
    .then(response => response.json())
    .then(fetchCards => {
      this.setState({
        originCards: fetchCards.mkCards,
        allCards: fetchCards.mkCards
      }, () => { this.getFromLocalStorage() });
    })
    .catch(error => {
      throw new Error(error);
    });
  }

  getFromLocalStorage = () => {
    if(localStorage.length > 0){
      const cardOrigins = this.state.allCards
      const correctArray = JSON.parse(localStorage.getItem('correctQuestions'));
      this.setState({
        correctCards: correctArray,
        localStorageExist: true
      }, () => { this.saveCardOrigins() });
    }

  }

  saveCardOrigins = () => {
    const originalCards = this.state.allCards;
    this.setState({
      originCards: originalCards
    }, () => { this.removeCorrectCards() })

  }

  removeCorrectCards = () => {
    console.log('afe', this.state.originCards);
    const { allCards, correctCards, originCards } = this.state;
    correctCards.forEach((correctCard) => {
      let foundCard = allCards.find(card => card.id === correctCard.id);
      let cardIndex = allCards.indexOf(foundCard);
      allCards.splice(cardIndex, 1);
    });
  }

  handleChangeCard = (clickedAnswer) => {
    const { allCards } = this.state;
    const currentCard = allCards[this.state.currCardIndex];
    let currIndex = this.state.currCardIndex;
    if(currentCard.correctAnswer === clickedAnswer){
      this.state.correctCards.push(currentCard);
      localStorage.setItem('correctQuestions', JSON.stringify(this.state.correctCards));
    }
    currIndex++;
    this.changeCardIndex(currIndex);
  }

  changeCardIndex = (currIndex) => {
    this.setState({
      currCardIndex: currIndex
    });
  }

  startQuiz = () => {
    this.setState({
      currentPlayer: true,
    })
  }

  resetQuiz = () => {
    localStorage.clear();
    this.setState({
      currentPlayer: false,
      localStorageExist: false,
    })

  }

  render() {
    const { allCards, currCardIndex, localStorageExist } = this.state;

    let displayContent;

    if(this.state.currentPlayer === false) {
      displayContent = <WelcomeScreen localStorageExist={localStorageExist}
                                      startQuiz={this.startQuiz} />
    } else {
      displayContent = <CardContainer currentCard={allCards[currCardIndex]}
                                      onChangeCard={this.handleChangeCard} />
    }

    console.log(this.state);
    return (
      <div className="App">
        <Header correctCards={this.state.correctCards}/>
        <main className="main-content">
          {displayContent}
        </main>
        <Footer resetQuiz={this.resetQuiz}
                currentPlayer={this.state.currentPlayer}
              />
      </div>
    )
  }
}

export default App;
