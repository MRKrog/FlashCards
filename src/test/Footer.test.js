import React from "react";
import { shallow } from "enzyme";
import App from "../App";
import Footer from "../Footer";

const resetQuiz = jest.fn();
const currentPlayer = App.currentPlayer;

describe("Footer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer resetQuiz={resetQuiz}
                              currentPlayer={currentPlayer}
      />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  
});
