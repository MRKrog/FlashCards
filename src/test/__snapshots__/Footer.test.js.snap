// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot when all data is passed correctly 1`] = `
<footer
  className="App-footer"
>
  <p>
    Â© 2019
  </p>
</footer>
`;

exports[`Footer should match snapshot when props currentPlayer is equal to true and show the reset game btn 1`] = `
<footer
  className="App-footer"
>
  <div
    className="BtnAnswer"
  >
    <button
      onClick={[MockFunction]}
    >
      Reset Game
    </button>
  </div>
</footer>
`;
